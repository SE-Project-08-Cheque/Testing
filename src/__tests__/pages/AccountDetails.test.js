// import dependencies
import React from "react";
// import react dom
import ReactDOM from "react-dom";
// import react-testing methods
import { render, cleanup, fireEvent } from "@testing-library/react";
// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";
// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import AccountDetails from "../../pages/Account_Details";
import { MemoryRouter } from "react-router-dom";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <AccountDetails 
        NIC= "970590054V"
        full_name= "Chalindu Kodikara"
        name_with_init = "K.A. Chalindu"
        dob= "1997/02/29"
      />
    </MemoryRouter>,
    div
  );
});

// it("renders Dashboard page correctly", () => {

//   const { getByTestId } = render(
//     <MemoryRouter>
//       <AccountDetails 
//         NIC= "970590054V"
        // full_name= "Chalindu Kodikara"
        // name_with_init = "K.A. Chalindu"
        // dob= "1997/02/29"
//       />
//     </MemoryRouter>
//   );
//   expect(getByTestId("user_type")).toHaveTextContent("Student");
//   expect(getByTestId("user_id")).toHaveTextContent("970590054V");
//   expect(getByTestId("username")).toHaveTextContent("Chalindu");
//   expect(getByTestId("account_no")).toHaveTextContent("123");
// });


it("matches snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <AccountDetails NIC= "970590054V"
        full_name= "Chalindu Kodikara"
        name_with_init = "K.A. Chalindu"
        dob= "1997/02/29" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

