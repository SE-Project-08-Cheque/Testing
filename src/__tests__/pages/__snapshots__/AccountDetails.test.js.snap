// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="dashboard"
>
  <div
    className="css-ure0nq"
  >
    <div
      className="css-dbpzvv"
    >
      <div
        className="css-1rul54"
      >
        <div
          className="css-gmuwbf"
        >
          <p
            className="chakra-text css-kbn7if"
          >
            
          </p>
        </div>
        <div
          className="css-sc5c3h"
        >
          <span
            className="chakra-badge css-ifsvay"
          >
            Online
          </span>
        </div>
        <div
          className="css-16n1mrt"
        >
          <div
            aria-valuemax={100}
            aria-valuemin={0}
            className="css-12om7f"
            data-indeterminate=""
            role="progressbar"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
      </div>
      <div
        className="css-8atqhb"
      >
        <div
          className="chakra-stack css-zsoya5"
        >
          <a
            href="/home"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="overview"
              type="button"
            >
              Overview
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/Account_Details"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="personal_details"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Personal Details
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/Contact_Details"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="contact_details"
              type="button"
            >
              Contact Details
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/AllNotifications"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="notification"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Notification
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <a
            href="/ForgotPin"
          >
            <button
              className="chakra-button css-35wvkn"
              data-testid="forgot_pin"
              type="button"
            >
              <div
                className="css-gmuwbf"
              >
                Fogot Pin
              </div>
            </button>
          </a>
          <div
            className="chakra-stack__divider css-1kh2tyg"
          />
          <hr
            aria-orientation="horizontal"
            className="chakra-divider css-1upb9tn"
            role="separator"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1d9x0xk"
  >
    <div
      className="css-1y2qylg"
    >
      <div
        className="css-gmuwbf"
      >
        <div
          className="css-rtmwg4"
        />
      </div>
      <div
        className="css-1orkoap"
      />
      <div
        className="css-9hhi3x"
      >
        <p
          className="chakra-text css-epm5mk"
        >
          <button
            aria-label="Send email"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.432,10.524C20.787,7.614,16.4,4.538,12,4.6,7.6,4.537,3.213,7.615.568,10.524a2.211,2.211,0,0,0,0,2.948C3.182,16.351,7.507,19.4,11.839,19.4h.308c4.347,0,8.671-3.049,11.288-5.929A2.21,2.21,0,0,0,23.432,10.524ZM7.4,12A4.6,4.6,0,1,1,12,16.6,4.6,4.6,0,0,1,7.4,12Z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="2"
                />
              </g>
            </svg>
          </button>
          NIC : 
          970590054V
                          
        </p>
      </div>
      <div
        className="css-15j5x0g"
      >
        <p
          className="chakra-text css-epm5mk"
        >
          <button
            aria-label="Call"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="5"
                />
                <path
                  d="M12 1v2"
                />
                <path
                  d="M12 21v2"
                />
                <path
                  d="M4.22 4.22l1.42 1.42"
                />
                <path
                  d="M18.36 18.36l1.42 1.42"
                />
                <path
                  d="M1 12h2"
                />
                <path
                  d="M21 12h2"
                />
                <path
                  d="M4.22 19.78l1.42-1.42"
                />
                <path
                  d="M18.36 5.64l1.42-1.42"
                />
              </g>
            </svg>
          </button>
          Full Name : 
          Chalindu Kodikara
        </p>
      </div>
      <div
        className="css-15j5x0g"
      >
        <p
          className="chakra-text css-qxtfng"
        >
          <button
            aria-label="Send email"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </button>
          Name with initials : 
          K.A. Chalindu
        </p>
      </div>
      <div
        className="css-15j5x0g"
      >
        <p
          className="chakra-text css-qxtfng"
        >
          <button
            aria-label="Send email"
            className="chakra-button css-1v9u08u"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
                fill="currentColor"
              />
            </svg>
          </button>
          Date of Birth : 
          1997/02/29
        </p>
      </div>
    </div>
  </div>
</div>
`;
